src/document.rs

```diff
@@ -282,6 +428,42 @@ impl fmt::Debug for Document {
     }
 }
 
+impl Serialize for Document {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map =
+            serializer.serialize_map(Some(self.field_values.len()))?;
+        for (k, v) in &self.field_values {
+            let ser_v: Vec<_> =
+                v.iter().map(BorrowedSerdeValue::from).collect();
+            map.serialize_entry(&k, &ser_v)?;
+        }
+        map.end()
+    }
+}
+
+impl<'de> Deserialize<'de> for Document {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        HashMap::<String, Vec<SerdeValue>>::deserialize(deserializer).map(
```

Comment:
I would recommend sticking to `BTreeMap` as the conversion from `HashMap` to `BTreeMap` will definitely need to re-allocate whereas `BTreeMap` could potentially avoid it.

Reply:
Oops, good catch, done.